### PR TITLE
#46 components to base repo

### PR DESCRIPTION
**Pull request by Chris-Petty**
related #46

## Change summary
Base components for the table. Except headers.

## Testing
Steps to reproduce or otherwise test the changes of this PR:
- [ ] run the demo app and play with it

### CODE REVIEW

File: src/CheckableCell.js [hunk 1]
```diff
@@ -1,98 +1,43 @@
...
+    disabled,
+    onCheckAction,
+    onUncheckAction,
+    dispatch,
+  }) => {
+    console.log(`- CheckableCell: ${rowKey},${columnKey}`)
+
+    const _onPressAction = isChecked ? onUncheckAction : onCheckAction
+
+    const _renderCheck = () =>
```
Comment (joshxg) [inline src/CheckableCell.js]: I think you'd use the same component, just a prop `checked` where styles change/component renders differently? And if you do that is this needed or can you just use `TouchableCell`?

File: src/EditableCell.js [hunk 2]
```diff
@@ -1,90 +1,94 @@
...
-  onEndEditing() {
-    // If the field is cleared, write null to property
-    const newValue = this.state.value === '' ? null : this.state.value;
-    this.props.onEndEditing(this.props.target, newValue);
-  }
+    // Too many TextInputs causes React Native to crash, so only
+    // truly mount the TextInput when the Cell is focused.
+    // Use TouchableWithoutFeedback because we want the appearance and
+    // feedback to resemble a TextInput regardless of focus.
+    if (!isFocused) {
```
[the remark this answers is not in the document]
Comment (Chris-Petty) [reply src/EditableCell.js]: Not implemented. You can dismiss the keyboard and or blur the textInput by tapping on a regular cell as usual but the last focused cell will keep render a text input until another editable cell is focused. That's all state management though, the cells aren't independently aware of each other.

File: src/CheckableCell.js (hunk 1 above)
Comment (Chris-Petty) [reply src/CheckableCell.js]: I was actually planning to go ham and make it have 4 component props:
- Checked...
- Unchecked...
- CheckedDisabled...
- UncheckedDisabled...

This is just a helper cell for a common use case in mobile. It should handle the 4 render states you could expect. You could equally make the argument that it's redundant along with the rest of this package, cause you can just implement with VirtualisedList, View, TextInput, TouchableOpacity, Text and flexibility of anything else. The base components are just some default styled sugar.

File: src/EditableCell.js [hunk 3]
```diff
@@ -1,90 +1,94 @@
...
+}
 
 const defaultStyles = StyleSheet.create({
   cell: {
     flex: 1,
+    backgroundColor: 'pink',
     justifyContent: 'center',
   },
-  text: {
-    right: -9, // This is to account for RN issue 1287, see https://github.com/facebook/react-native/issues/1287
```
Comment (Chris-Petty) [inline src/EditableCell.js]: might be worth holding onto this!

File: src/TouchableCell.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    CellComponent,
+    ...otherProps
+  }) => {
+    console.log(`- TouchableCell: ${rowKey},${columnKey}`)
+
+    const onPress = () => {
+      dispatch(onPressAction(rowKey, columnKey))
+    }
+
+    const Container = CellComponent ? CellComponent : TouchableOpacity
```
[the remark this answers is not in the document]
Comment (Chris-Petty) [reply src/TouchableCell.js]: Yee could work

File: src/TouchableCell.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    dispatch,
+    children,
+    renderChildren,
+    CellComponent,
+    ...otherProps
+  }) => {
+    console.log(`- TouchableCell: ${rowKey},${columnKey}`)
+
+    const onPress = () => {
+      dispatch(onPressAction(rowKey, columnKey))
```
[the remark this answers is not in the document]
Comment (Chris-Petty) [reply src/TouchableCell.js]: The point of dispatch is that the reference never changes, so using a closure you're gonna have to memoize it, at which point it's just redundant I think. Unless you mean to use it between the component and props, at which point you're a fraction away from `redux.connect()` 😄 

I'd be keen to see an example of how you'd do it in this case. I could be thinking about it wrong, but I rarely have a case where I thought closures would reduce complexity. But that could be in part cause they are slightly more conceptually complex than fundamental patterns and I've rarely used them to get familiar.

**Pull request closed without merging**